src/utils/storage.js

```diff
@@ -13,43 +10,65 @@ const getBlankUserWordList = () => ({
   five: {},
 });
 
+export const getAllUsers = async () => {
+  try {
+    const usersJson = AsyncStorage.getItem('@users').then(JSON.parse);
+    const users = usersJson || {};
+    return users;
+  } catch {
+    return {};
+  }
+};
+
 export const getUser = async () => {
   try {
     const user = await AsyncStorage.getItem('@user');
-    if (user !== null) {
-      return user;
-    }
-    return null;
+    return user;
   } catch (e) {
     return null;
   }
 };
 
+export const addUser = async (user) => {
+  try {
+    const users = await getAllUsers();
+    users[user] = { wordList: getBlankUserWordList() };
+    await AsyncStorage.setItem('@users', JSON.stringify(users));
+    return true;
+  } catch {
+    return false;
+  }
+};
+
 export const setUser = async (user) => {
   try {
     await AsyncStorage.setItem('@user', user);
+    const users = await getAllUsers();
+    if (!users[user]) {
+      addUser(user);
+    }
     return true;
   } catch {
     return false;
   }
 };
 
 export const getUserData = async (userParam) => {
-  let user;
-  if (!userParam) {
-    user = await getUser();
-  } else {
-    user = userParam;
-  }
+  const user = userParam || (await getUser());
```

Comment:
Could this be problematic? I can't think of any problems this would cause but my instincts say this might be an issue 🤔

Reply:
This seem fine to me....

Just so I'm following this structure:

- `@user` is the username of the current user
- `@users` is an object where the keys are usernames and the value is `{ wordlist: { one, two...} }`?

Reply:
Yes that's right!

Reply:
Gotcha! In this case, I'd be tempted to name `user` as `currentUserName`, but that's a pretty minor detail.

Reply:
Gotcha. I think that's very fair and probably a good idea! But given the timeframe I don't see myself getting to it this week.